### PR TITLE
fix: request ACL for ACO [#576]

### PR DESCRIPTION
**Pull request by vitshev**
https://nda.ya.ru/t/x52D79jM767p6v
    
https://github.com/ytsaurus/ytsaurus-ui/issues/576

### CODE REVIEW

**Review by SimbiozizV — DISMISSED** [review]

**Review by ma-efremoff — CHANGES_REQUESTED** [review]
please do not merge the PR

File: packages/ui/src/shared/utils/index.ts [hunk 1]
```diff
@@ -48,3 +48,12 @@ export function isYTError(error: any): error is YTError {
     const {yt_javascript_wrapper} = (error as YTError) ?? {};
     return Boolean(yt_javascript_wrapper?.xYTTraceId && yt_javascript_wrapper?.xYTRequestId);
 }
+
+export function makeRegexpFromSettings(value?: string) {
+    console.log('makeRegexpFromSettings', value);
```
Comment (ma-efremoff) [inline packages/ui/src/shared/utils/index.ts:53]: Please remove `console.log(...)`

File: packages/ui/src/ui/utils/acl/index.ts [hunk 2]
```diff
@@ -53,7 +55,10 @@ export function normalizeIdmParams(idmKind: IdmKindType, path = '') {
                 idmKind: IdmObjectType.TABLET_CELL_BUNDLE,
                 path: path.substr('//sys/tablet_cell_bundles/'.length),
             };
-        } else if (ACCESS_CONTROL_OBJECT_ANY_REGEXP.test(path)) {
+        } else if (
+            ACCESS_CONTROL_OBJECT_ANY_REGEXP &&
+            ACCESS_CONTROL_OBJECT_ANY_REGEXP.test(path)
```
Comment (ma-efremoff) [inline packages/ui/src/ui/utils/acl/index.ts:60]: `ACCESS_CONTROL_OBJECT_ANY_REGEXP?.test(path)`

File: packages/ui/src/ui/utils/acl/index.ts [hunk 3]
```diff
@@ -17,9 +19,9 @@ const POOL_TREES_REGEXP = new RegExp('//sys/pool_trees/[^/]+/.+');
 const TCB_REGEXP = new RegExp('//sys/tablet_cell_bundles/.+');
 
 // //sys/access_control_object_namespaces...
-const ACCESS_CONTROL_OBJECT_ANY_REGEXP = new RegExp(
-    '//sys/access_control_object_namespaces[^/+]{0,}',
-);
+const ACCESS_CONTROL_OBJECT_ANY_REGEXP =
```
Comment (ma-efremoff) [inline packages/ui/src/ui/utils/acl/index.ts:22]:
```ts
const ACCESS_CONTROL_OBJECT_ANY_REGEXP = makeRegexpFromSettings(uiSettings.reUseEffectiveAclForPath);
```

File: packages/ui/src/ui/utils/acl/acl-api.ts [hunk 4]
```diff
@@ -53,9 +53,7 @@ export function getAcl({
     sysPath,
 }: GetAclParams): Promise<PreparedAclData> {
     const api = UIFactory.getAclApi();
-    if (kind === IdmObjectType.UI_EFFECTIVE_ACL) {
-        return getEffectiveAcl(sysPath);
-    }
```
Comment (ma-efremoff) [inline packages/ui/src/ui/utils/acl/acl-api.ts:58]: please bring back the if-expression

File: packages/ui/src/server/configs/common.ts [hunk 5]
```diff
@@ -58,6 +58,7 @@ const config: Partial<AppConfig> = {
         reHashFromNodeVersion: '[^~]+~(?<hash>[^+]+)',
         directDownload: true,
         docsBaseUrl: process.env.YT_DOCS_BASE_URL || 'https://ytsaurus.tech/docs/en',
+        reUseEffectiveAclForPath: '//sys/access_control_object_namespaces[^/+]{0,}',
```
Comment (ma-efremoff) [inline packages/ui/src/server/configs/common.ts:61]: I don't think we need default value for the option here (or in local.ts). But it is a good idea to have corresponding e2e to be sure the functionality works propertly.

File: packages/ui/src/ui/constants/acl.ts [hunk 6]
```diff
@@ -69,7 +69,20 @@ export type PermissionSettings = {
 };
 
 export const PERMISSIONS_SETTINGS: Record<IdmKindType, PermissionSettings> = {
-    [IdmObjectType.UI_EFFECTIVE_ACL]: {...PATH_SETTINGS, permissionsToRequest: []},
+    [IdmObjectType.UI_EFFECTIVE_ACL]: {
```
[the remark this answers is not in the document]
Comment (ma-efremoff) [reply packages/ui/src/ui/constants/acl.ts]: I think it is enough to check that some acl are rendered and there are no buttons to request/manage them.

File: packages/ui/src/ui/utils/acl/index.ts (hunk 3 above)
Comment (ma-efremoff) [reply packages/ui/src/ui/utils/acl/index.ts:22]: According to current implementation of `makeRegexpFromSettings` you should get `undefined`

Comment (ma-efremoff) [reply packages/ui/src/ui/utils/acl/index.ts:22]: Oh, sorry, I am wrong. Can we rewrite `makeRegexpFromSettings` to have `undefined` for the case?

File: packages/ui/src/shared/utils/index.ts [hunk 7]
```diff
@@ -48,3 +48,11 @@ export function isYTError(error: any): error is YTError {
     const {yt_javascript_wrapper} = (error as YTError) ?? {};
     return Boolean(yt_javascript_wrapper?.xYTTraceId && yt_javascript_wrapper?.xYTRequestId);
 }
+
+export function makeRegexpFromSettings(value?: string) {
+    try {
+        return new RegExp(value!);
```
Comment (ma-efremoff) [inline packages/ui/src/shared/utils/index.ts:54]: it is better to return undefined when `Boolean(value) === false`

File: packages/ui/src/server/configs/e2e/local.ts [hunk 8]
```diff
@@ -43,6 +43,7 @@ const e2eConfig: Partial<AppConfig> = {
 
         reUnipikaAllowTaggedSources: ['^https://yastatic\\.net/'],
         hideReferrerUrl: 'https://h.yandex-team.ru/',
+        reUseEffectiveAclForPath: '//sys/access_control_object_namespaces[^/+]{0,}',
```
Comment (ma-efremoff) [inline packages/ui/src/server/configs/e2e/local.ts:46]: Editing of the file is a good reason to edit some files from `packages/UI/tests`.

File: packages/ui/tests/e2e/pages/aco.spec.ts [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    const acoPage = new ACOPage({page});
+
+    await acoPage.open();
+
+    await page.evaluate(() => {
+        //@ts-expect-error
+        window.__DATA__.uiSettings.reUseEffectiveAclForPath = undefined;
+    });
+
+    await acoPage.open();
```
Comment (ma-efremoff) [inline packages/ui/tests/e2e/pages/aco.spec.ts:30]: Do we really need secondary open-call?

File: packages/ui/tests/e2e/pages/aco.spec.ts [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+});
+
+test('@ACO: the permissions request button should be hidden', async ({page, context}) => {
+    const acoPage = new ACOPage({page});
+
+    await acoPage.open();
+    await acoPage.waitForTableLoad();
+    await acoPage.clickOnACLTab();
+
+    await expect(page.getByText('Request permissions')).toBeHidden();
```
Comment (ma-efremoff) [inline packages/ui/tests/e2e/pages/aco.spec.ts:43]: We need to wait content of ACL tab before check visibility of the button.

**Review by ma-efremoff — APPROVED** [review]

**Pull request merged**